test/simplifyExpression/distributeSearch/distributeSearch.test.js

```diff
@@ -69,5 +64,21 @@ describe('distribute with fractions', function () {
     ],
   ];
 
-  tests.forEach(t => testDistributeResult(t[0], t[1]));
+  const multiStepTests = [
+
+    ['(2 / x +  3x^2) * (x^3 + 1)',
+      ['(2 * (x^3 + 1) / x + 3x^2 * (x^3 + 1))',
```

Comment:
yeah, that's what I was thinking too. You can explicitly parenthesize it by wrapping it in a `Node.Creator.Parenthesis` node, I hope that fixes this